tests/test_github_api.py

```diff
@@ -12,8 +12,7 @@
 from ai_nexus_backend import github_api
 
 
-class Test_AssembleReadmeEndpointFromRepoUrl:
-    """Regex pattern testing for internal utility."""
+class TestGithubClient:
```

Comment:
I'd prefer to see some unit tests rather than functional tests, as this will rely on some setup in CI to be able to access the github API, which is probably unnecessary. I'd treat these more as one off ad hoc tests rather than something we'd run automatically.

Reply:
Hi @townxelliot  - everything in tests should be completely isolated - unless I didn't spot something. I have the _test_cases urls simply for regex matching tests. No api queries should be happening. Did you spot that anywhere?

Reply:
Just to confirm (doubting my own sanity) -  ` client_fixture._assemble_readme_endpoint_from_repo_url()`
This internal utility will not query the endpoint. It just does some regex matching and string manipulation to get the endpoint url we need from the github repo url.

If this isn't clear, I should go ahead and add some inline comments to help though. What do you think?